lib/components/base_component.dart

```diff
@@ -0,0 +1,153 @@
+import 'dart:ui';
+
+import 'package:meta/meta.dart';
+import 'package:ordered_set/comparing.dart';
+import 'package:ordered_set/ordered_set.dart';
+
+import '../effects/effects.dart';
+import '../effects/effects_handler.dart';
+import '../extensions/vector2.dart';
+import '../game.dart';
+import '../text_config.dart';
+import 'component.dart';
+
+/// This can be extended to represent a basic Component for your game.
+///
+/// The difference between this and [Component] is that the [BaseComponent] can
+/// have children, handle effects and can be used to see whether a position on
+/// the screen is on your component, which is useful for handling gestures.
+abstract class BaseComponent extends Component {
+  final EffectsHandler _effectsHandler = EffectsHandler();
+
+  final OrderedSet<Component> _children =
+      OrderedSet(Comparing.on((c) => c.priority));
+
+  OrderedSet<Component> get children =>
+      OrderedSet<Component>()..addAll(_children);
```

Comment:
should we expose a clone method on ordered_set?

Reply:
And a reorder method would be very useful too